src/net_processing.cpp

```diff
@@ -2779,7 +2779,9 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         }
         m_addrman.Add(vAddrOk, pfrom.addr, 2 * 60 * 60);
         if (vAddr.size() < 1000) peer->m_getaddr_sent = false;
```

Comment:
Does this work as intended though? 
1. Alice connects to Bob, each other send GETADDR, m_getaddr_sent=true
2. Alice self-announces, Bob sets `m_getaddr_sent=false`
3. Alice responds to GETADDR, `m_getaddr_sent` is already false, so the check relying on it is not really correct.

The check being incorrect seems to be not critical because `vAddr.size() <= 10` is very likely to be false anyway (assuming sane addrman), but what's the point then?

What this check actually does is `don't relay the first self-announcement`. I don't think this is intended? Or if it is, it's very unclear from the codebase esp with that variable name. We could really benefit from a comment.

Reply:
Note: what i'm asking here is not really related to the change of this PR.

Reply:
I agree, it doesn't work as intended! There was some earlier discussion about this (the flag was recently renamed from `fGetAddr`), see https://github.com/bitcoin/bitcoin/pull/21707#issuecomment-824361280 and you may recall #19794. Maybe it would be good to revisit this topic...